samples/common/src/image_rendering.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+use image::{imageops::FilterType, DynamicImage};
+use sdl2::pixels::{Color, PixelFormatEnum};
+use sdl2::rect::Rect;
+use sdl2::render::WindowCanvas;
+use sdl2::surface::Surface;
+use sdl2::Sdl;
+
+// This module is based on SDL2 (Simple DirectMedia Layer). It complements the sdl2 crate by providing
+// methods that make it easier to render images.
+
+/// Create a canvas with an enclosing window.
+///
+/// # Arguments
+/// * `sdl_context` - The SDL context.
+/// * `window_title` - The window's title.
+/// * `window_width` - The window's width.
+/// * `window_height` - The window's height.
+pub fn create_canvas(
+    sdl_context: &mut Sdl,
+    window_title: &str,
+    window_width: u32,
+    window_height: u32,
+) -> Result<WindowCanvas, String> {
```

Comment:
for rust libraries it's generally not recommended to use strings for the error type. this makes it hard to do anything but log the error. I think other places in Ibeji are using `Status` (e.g. [here](https://github.com/eclipse-ibeji/ibeji/blob/3073a3141c25057f901233a980b29a4903bf37f5/samples/mixed/provider/src/provider_impl.rs#L63)) so ideally we should use that here too

Reply:
I used String to be consistent with the sdl2 crate's methods, which use String.

Status comes from the tonic crate and is currently only used in conjunction with gRPC-related calls.

Reply:
> I used String to be consistent with the sdl2 crate's methods, which use String.

Converting errors can be done with `map_err`. It's also worth mentioning that the library owner would prefer to use a proper error type but hasn't made the change due to concerns with breaking changes: https://github.com/Rust-SDL2/rust-sdl2/issues/1053

> Status comes from the tonic crate and is currently only used in conjunction with gRPC-related calls.

Then `Status` wouldn't be the right thing to use, we should find something else. Consider defining your own error type similar to what Freyja does, or you could use `anyhow` (not recommended for libraries but easy to use and better than `String`) or `thiserror`, or even reuse `std::io::Error`